OpenRA.Game/Network/Session.cs

```diff
@@ -90,11 +90,16 @@ public string FirstEmptyBotSlot()
 			return Slots.FirstOrDefault(s => !s.Value.Closed && ClientInSlot(s.Key) == null && s.Value.AllowBots).Key;
 		}
 
-		public bool IsSinglePlayer
+		public bool IsSingleHuman
```

Comment:
IMO this property should go away.  A lot of code uses this as a proxy for "one human player", which is wrong, because this will return false if there is one human player plus another human observer.

I suggest introducing an extra `NonBotClients` property that works like `HumanPlayers`/`NonBotPlayers` but doesn't filter spectators.  We can then replace all the uses of `IsSinglePlayer` with the appropriate version, using `.Count()` inline instead of introducing more mostly redundant methods.